Lib/test/_test_atexit.py

```diff
@@ -135,6 +135,34 @@ def func():
         finally:
             atexit.unregister(func)
 
+    def test_eq_unregister_clear(self):
+        # Issue #112127: callback's __eq__ may call unregister or _clear
+        cnt = 0
+        class Func:
+            def __init__(self, action, eq_ret_val):
+                self.action = action
+                self.eq_ret_val = eq_ret_val
+
+            def __call__(self):
+                return
+
+            def __eq__(self, o):
+                cnt += 1
+                if cnt == 1:
+                    self.action(o)
+                return self.eq_ret_val(o)
+
+        for action in lambda o: atexit.unregister(self), \
+                lambda o: atexit.unregister(o), lambda o: atexit._clear():
```

Comment:
It is better to add explicit parentheses and place each item on a separate line.